### PR TITLE
Yatai deployment service implementation WIP

### PR DESCRIPTION
**Pull request by parano**

### CODE REVIEW

File: bentoml/deployment/store.py [hunk 1]
```diff
@@ -16,45 +16,92 @@
...
 
     id = Column(String, primary_key=True)
     name = Column(String, unique=True)
-    namespace = Column(String)
+    namespace = Column(String, nullable=False)
 
     spec = Column(JSON)
-    labels = Column(JSON)
-    annotation = Column(JSON)
+    labels = Column(JSON, nullable=False, default={})
```
Comment (yubozhao) [inline bentoml/deployment/store.py]: I think labels and annotation fields are optional, correct?  If they are optional, then nullable won't need to set to be False

File: bentoml/deployment/store.py [hunk 2]
```diff
@@ -16,45 +16,92 @@
...
+        labels=MessageToDict(deployment_pb.labels),
+        annotation=MessageToDict(deployment_pb.labels),
+    )
+
+
+def deployment_orm_obj_to_pb(deployment_obj):
+    return deployment_pb2.Deployment(
+        name=deployment_obj.name,
+        namespace=deployment_obj.namespace,
+        spec=ParseDict(deployment_obj.spec, deployment_pb2.DeploymentSpec),
```
Comment (yubozhao) [inline bentoml/deployment/store.py]: I test this in ipython, and got 

```
AttributeError: 'google.protobuf.pyext._message.FieldProperty' object has no attribute 'SetInParent'
```
I got this error, because inside DeploymentSpec, we have additional messages like `SageMakerOperatorConfig` and others. We probably want to make sure parse those config with `ParseDict` first

File: bentoml/deployment/store.py (hunk 1 above)
Comment (parano) [reply bentoml/deployment/store.py]: Will need to test out how it works when doing filtering by label on null jsonb filed in postgres

Comment (parano) [reply bentoml/deployment/store.py]: they are indeed optional, note the default value '{}'

File: bentoml/deployment/store.py (hunk 2 above)
Comment (parano) [reply bentoml/deployment/store.py]: oh thank for catching this, that was a typo, should be ```ParseDict(deployment_obj.spec, deployment_pb2.DeploymentSpec())```
